src/client.ts

```diff
@@ -351,10 +351,11 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
   }
 
   public async setUser(userInfo: UserInfo): Promise<void> {
+    // The JS SDK will try to assign VUID if the user ID is null
     this.setCurrentUserContext(userInfo);
 
     this.user = {
-      id: userInfo.id || DefaultUser.id,
+      id: userInfo.id || this.getUserContext()?.getUserId() || DefaultUser.id,
```

Comment:
If userInfo.id is not provided, should we re-use previous userId or it is intended that we go back to the default it that is vuid?

Reply:
Thanks for asking @raju-opti . If `userInfo.id` is not provided, in client-side cases, we're looking for the `vuid_385d...` to be issued by `this.setCurrentUserContext(userInfo);`.  In server-side context, yes, we're back to default id.

Reply:
I was asking more about cases like:

`
setUser({ id: 'mike-chu' });
// do bunch of other stuff
// then
setUser({})
`
In that case the user will want to clear the userId, but this will just pull the previous userId ('mike-chu') instead will logically changes nothing.

Reply:
Would `setCurrentUserContext` from client.tsL330 be hit?  

Let's pair up maybe on Teams?